third_party/move/tools/move-linter/src/utils.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module holds utility functions for the Move linter.
+
+use move_model::ast::{ExpData, Operation};
+
+/// Returns `true` if two expressions represent the same simple access pattern.
+/// This compares nested `Select`, `Borrow`, and `Deref` calls,
+/// local variable references, and literal values for structural equality.
+/// Patterns that use global storage or non-builtin function (including vector operations)
+/// are not considered simple access patterns for the purpose of this function and return false.
+pub(crate) fn is_simple_access_equal(expr1: &ExpData, expr2: &ExpData) -> bool {
+    match (expr1, expr2) {
+        (ExpData::Call(_, op1, args1), ExpData::Call(_, op2, args2)) => {
+            op1 == op2
+                && matches!(
+                    op1,
+                    Operation::Select(_, _, _)
+                        | Operation::Borrow(_)
+                        | Operation::Deref
+                        | Operation::SelectVariants(_, _, _)
+                )
+                && args1.len() == args2.len()
```

Comment:
Is this check `args1.len() == args2.len()` necessary?

Reply:
Nice catch.